### PR TITLE
Optional recoverability flag on recipe components

### PR DESCRIPTION
**Pull request by macrosblackd**
Adds the ability to flag a component as unrecoverable when disassembling.
This doesn't overwrite the UNRECOVERABLE flag that certain items already
have.

Updated the documentation with the unrecoverability information.

### CODE REVIEW

**Comment by KA101** [comment]
Care to apply this to a component or two?  Would be helpful to see the use case.

**Comment by macrosblackd** [comment]
you can see them in archery tweaks. this is something I added and committed separately.

File: doc/JSON_INFO.md [hunk 1]
```diff
@@ -216,10 +216,10 @@ The syntax listed here is still valid.
   [ "fur", 1 ]
 ],
 [
-  [ "plant_fibre", 20 ],
-  [ "sinew", 20 ],
-  [ "thread", 20 ],
-  [ "duct_tape", 20 ]
+  [ "plant_fibre", 20, false ], // Optional flag for recoverability, default is true.
```
Comment (narc0tiq) [inline doc/JSON_INFO.md]: Could the flag be some kind of string like "NO_RECOVER", instead? False doesn't exactly tell me what I'm looking at here, unless I read the docs (which nobody ever does).

Comment (KA101) [inline doc/JSON_INFO.md]: Yeah, I saw the falses in the archery PR and thought they were related to this.  But that's because I saw this first.

**Pull request merged**